lib/msal-common/src/unifiedCache/UnifiedCacheManager.ts

```diff
@@ -61,47 +111,341 @@ export class UnifiedCacheManager {
     }
 
     /**
-     * Returns if the account is in Cache
+     * saves account into cache
+     * @param account
+     */
+    saveAccount(account: AccountEntity): void {
+        const cache = this.getCacheInMemory();
+        const key = account.generateAccountKey();
+        cache.accounts[key] = account;
+        this.setCacheInMemory(cache);
+    }
+
+    /**
+     * saves credential - accessToken, idToken or refreshToken into cache
+     * @param credential
+     */
+    saveCredential(credential: Credential): void {
+        const cache = this.getCacheInMemory();
+        const key = credential.generateCredentialKey();
+
+        switch (credential.credentialType) {
+            case CredentialType.ID_TOKEN:
+                cache.idTokens[key] = credential as IdTokenEntity;
+                break;
+            case CredentialType.ACCESS_TOKEN:
+                cache.accessTokens[
+                    key
+                ] = credential as AccessTokenEntity;
+                break;
+            case CredentialType.REFRESH_TOKEN:
+                cache.refreshTokens[
+                    key
+                ] = credential as RefreshTokenEntity;
+                break;
+            default:
+                console.log("Cache entity type mismatch");
+        }
+
+        this.setCacheInMemory(cache);
+    }
+
+    /**
+     * Given account key retrieve an account
+     * @param key
+     */
+    getAccount(key: string): AccountEntity {
+        return this.getCacheInMemory().accounts[key] || null;
+    }
+
+    /**
+     * retrieve a credential - accessToken, idToken or refreshToken; given the cache key
+     * @param key
+     */
+    getCredential(key: string): Credential {
+        const cache = this.getCacheInMemory();
+        switch (
+            key.split(Separators.CACHE_KEY_SEPARATOR)[
+                CacheKeyPosition.CREDENTIAL_TYPE
+            ]
+        ) {
+            case "idtoken":
+                return cache.idTokens[key] || null;
+            case "accesstoken":
+                return cache.accessTokens[key] || null;
+            case "refreshtoken":
```

Comment:
Can you put these in an enum?

Reply:
Looks like it might already exist?

Reply:
Need to convert existing ones to lower case. Done now.